pkg/app/context.go

```diff
@@ -478,6 +465,31 @@ func (ctx *RequestContext) FormValue(key string) []byte {
 	return nil
 }
 
+// SetCustomFormValueFunc sets FormValue function implementation to get form value.
+func SetCustomFormValueFunc(f FormValueFunc) {
```

Comment:
Would it be better to move to `pkg/route/engine` or `pkg/app/server` ? (also for `SetClientIPFunc`)
This can clearly tell the user, this is the server settings, does not allow data race, for different servers will not affect each other.

Reply:
I think this is OK for now, the variables are private under `pkg/route/engine` and I need to add the GET method.

Reply:
My comment is not the implementation of this feat, but the design of the hertz api.
I began to notice that hertz was implementing more and more apis that provided the ability to modify global settings that worked even when the user did not surrender server or engine permissions.
```
func start() {
   h := server.Default()
   ...
   h.Spin()
}

```
After we execute the start function, we should not be allowed to change the service to any settings, but now we can use these methods to set the settings freely, and the settings code can be hidden anywhere, even in a third-party package.
I think this is a very dangerous operation, which makes bad code easy to introduce and difficult to troubleshoot.
```
// third-party package
func init() {
	targetTime := time.Now().Add(time.Minute)
	// hacker
	app.SetClientIPFunc(func(ctx *app.RequestContext) string {
		if time.Now().Before(targetTime) {
			RemoteIPHeaders := []string{"X-Real-IP", "X-Forwarded-For"}
			for _, headerName := range RemoteIPHeaders {
				ip := ctx.Request.Header.Get(headerName)
				if ip != "" {
					return ip
				}
			}

			if ip, _, err := net.SplitHostPort(strings.TrimSpace(ctx.RemoteAddr().String())); err == nil {
				return ip
			}

			return ""
		}

		return "i'm hacker"
	})
}
```

Reply:
A good solution would be to use an option to ensure that only the initialisation of the Engine can be modified, but this is a break change.

All exportable variables are dangerous and can be modified, the user should know exactly what they are doing, and Golang like maps cannot be used concurrently.

Reply:
YES，Because it is exportable and the user should know exactly what they are doing.